src/metabase/email/messages.clj

```diff
@@ -40,13 +43,42 @@
          (u/is-email? email)
          (string? hostname)
          (string? password-reset-url)]}
-  (let [message-body (html [:html
-                            [:body
-                             [:p (str (format "You're receiving this e-mail because you or someone else has requested a password for your user account at %s. " hostname)
-                                      "It can be safely ignored if you did not request a password reset. Click the link below to reset your password.")]
-                             [:p [:a {:href password-reset-url} password-reset-url]]]])]
+  (let [message-body (->> {:hostname hostname
+                           :passwordResetUrl password-reset-url}
+                          (stencil/render-file "metabase/email/password_reset"))]
     (email/send-message
      :subject      "[Metabase] Password Reset Request"
      :recipients   [email]
      :message-type :html
      :message      message-body)))
+
+;; HACK: temporary workaround to postal requiring a file as the attachment
+(defn- write-byte-array-to-temp-file
+  [img-bytes]
+  (let [file (java.io.File/createTempFile "metabase_pulse_image_" ".png")
+        fos (new java.io.FileOutputStream file)]
+    (.deleteOnExit file)
+    (.write fos img-bytes)
+    (.close fos)
+    file))
+
+(defn render-pulse-email
+  "Take a pulse object and list of results, returns an array of attachment objects for an email"
+  [pulse results]
+  (let [images (atom [])
+        render-img (fn [bytes] (reset! images (conj @images bytes)) (str "cid:IMAGE_" (-> @images count dec)))
+        body (apply vector :div (mapv (partial render-pulse-section render-img) results))
+        data-quote (rand-nth q/quotations)
```

Comment:
from a code formatting standpoint we have been aligning the values in cases like this, so it should be ...

```
(let [images     (atom [])
      render-img (fn [bytes] (reset! images (conj @images bytes)) (str "cid:IMAGE_" (-> @images count dec)))
      body       (apply vector :div (mapv (partial render-pulse-section render-img) results))
      data-quote (rand-nth q/quotations)]
  ...)
```